### PR TITLE
Sam/nix parallel

### PR DESCRIPTION
**Pull request by samrose**
## What kind of change does this PR introduce?

This draft PR introduces a "passive" experimental parallel build of postgresql 15, 16, experimental orioledb-16 all with known working extensions and wrappers. It will also include gh action workflows that will build and test all these variants, extensions and wrappers. Once those actions are added, I'll switch it from draft to actual PR.

Once this PR is merged, this portion of the project will just update and build in the background,  in an attempt to stabilize this work here over time. We won't try to immediately use any of it for build and deployment of production postgres, exts, and wrappers until we see it stabilize here for some time.

### CODE REVIEW

File: nix/ext/citus.nix
Comment (olirice) [inline nix/ext/citus.nix]: do we still need this one?

File: nix/ext/hypopg.nix
Comment (olirice) [inline nix/ext/hypopg.nix:5]: any way we can connect this up to values in ansible var file here https://github.com/supabase/postgres/blob/develop/ansible/vars.yml ?

it doesn't necessarily need to be in this PR but it would be great to make everything updatable from a single location

File: nix/ext/pg_tle.nix
Comment (olirice) [inline nix/ext/pg_tle.nix]: lets update the maintainers list

File: nix/ext/pgvector.nix
Comment (olirice) [inline nix/ext/pgvector.nix]: bump to 0.6.0

**Review by olirice — CHANGES_REQUESTED** [review]
looks great. none of the comments above are blockers to merging if you'd rather get this in and then update in smaller PRs but can we add the CI jobs before merging?

I also only just realized this was a draft so sorry for reviewing before it's ready!

File: nix/ext/citus.nix
Comment (samrose) [reply nix/ext/citus.nix]: @olirice awesome,  thanks for reviewing this! I put this together while I was stuck for some hours in CDG Paris Airport :-D

I'll just address your feedback in some commits and get the actions integrated here ASAP, and then we could merge it with all that feedback plus actions migrated over. So I will just start resolving this all now.

Comment (samrose) [reply nix/ext/citus.nix]: This one is resolved with https://github.com/supabase/postgres/pull/909/commits/73bee62e6ec161055ba75056e9c9a4585f2cadc9

File: nix/ext/pg_tle.nix
Comment (samrose) [reply nix/ext/pg_tle.nix]: Resolved with 651e1a2

File: nix/ext/hypopg.nix
Comment (samrose) [reply nix/ext/hypopg.nix:5]: @olirice  I am going to look into consuming that file, possibly matching on the extname, then using the version, etc, and so then being able to do the automation you describe.  I'll follow up with another PR that focuses just on that bit. Thanks for this suggestion, it's a great solution to integration with deployment.

File: nix/ext/pgvector.nix
Comment (samrose) [reply nix/ext/pgvector.nix]: resolved with https://github.com/supabase/postgres/pull/909/commits/28fa350acf5f791e7efedf7c45ed31c428872807

File: nix/ext/hypopg.nix
Comment (samrose) [reply nix/ext/hypopg.nix:5]: Will resolve with the plan to create an implementation above

**Comment by samrose** [comment]
@olirice wanted to ping and let you know that now our nix gh action workflows are integrated and passing (dropped justfile requirement too)

**Review by olirice — APPROVED** [review]

**Comment by samrose** [comment]
I have a bit more work to do on this one, to get this working with deployment files etc. But we decided just to keep rolling with this one, and have it all reviewed at once.

**Comment by samrose** [comment]
closing in favor of https://github.com/supabase/postgres/pull/925

**Pull request closed without merging**
